state/state.go

```diff
@@ -141,9 +153,55 @@ func detectRuntimeState(r *Runtime) error {
 			}
 		}
 	}
+	// This is for LVM mounted partitions which ghw cant find by itself
+	if !r.Recovery.Found || !r.OEM.Found {
+		detectLVMPartitions(r)
+	}
 	return nil
 }
 
+func detectLVMPartitions(r *Runtime) {
+	// Check if the oem is there
+	// lsblk works both mounted and unmounted, so it should work everywhere
+	if !r.OEM.Found {
+		out, err := utils.SH("lsblk /dev/disk/by-label/COS_OEM -o PATH,FSTYPE,MOUNTPOINT,SIZE,RO,LABEL -J")
+		mnt := &Lsblk{}
+		if err == nil {
+			err = json.Unmarshal([]byte(out), mnt)
+			// This should not happen, if there were no targets, the command would have returned an error, but you never know...
+			if err == nil && len(mnt.BlockDevices) == 1 {
+				r.OEM.Found = true
+				r.OEM.Name = mnt.BlockDevices[0].Path
```

Comment:
maybe this makes no significant difference but would it make sense to put the result of `mnt.BlockDevices[0]` in a variable?

Reply:
done